File: MercadeoUCAB-backend/entidades/src/main/java/ucab/dsw/entidades/Usuario.java
package ucab.dsw.entidades;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.util.List;

@Entity
@Table( name = "usuario" )
public class Usuario extends EntidadBase {

    //Columnas

    @Column( name = "nombreUsuario" )
    private String _nombre;
    
    @Column( name = "correo" )
    private String _correoelectronico;

    @Column( name = "codigoRecuperacion" )
    private String _codigoRecuperacion;

    //Relaciones
   
    @ManyToOne
    @JoinColumn( name = "fk_rol" )
    private Rol _rol;
    
    ////Getters, Setters, y otros metodos.

    public String getNombre() {
        return _nombre;
    }

    public String getCorreoelectronico() {
        return _correoelectronico;
    }

    public String getCodigoRecuperacion() {
        return _codigoRecuperacion;
    }

    public Rol getRol() {
        return _rol;
    }

    public void setNombre(String _nombre) {
        this._nombre = _nombre;
    }

    public void setCorreoelectronico(String _correoelectronico) {
        this._correoelectronico = _correoelectronico;
    }

    public void setCodigoRecuperacion(String _codigoRecuperacion) {
        this._codigoRecuperacion = _codigoRecuperacion;
    }

    public void setRol(Rol _rol) {
        this._rol = _rol;
    }

    public Usuario (String estatus) {
        super(estatus);
    }

    public Usuario( long id )
    {
        super( id );
    }

    public Usuario() {
        super();
    }

    @Override
    public String toString() {
        return "Usuario{" +
                "_nombre='" + _nombre + '\'' +
                ", _correoelectronico='" + _correoelectronico + '\'' +
                ", _codigoRecuperacion='" + _codigoRecuperacion + '\'' +
                ", _rol=" + _rol +
                '}';
    }
}

File: MercadeoUCAB-frontend/src/app/modelos/usuario.ts
import { Rol, Rol2 } from './rol';

export class Usuario {
    constructor(
       public _id: number,
       public _nombre: string,
       public _correoelectronico: string,
       public _token: string,
       public _estatus: string,
       public _rol: Rol
     ){}
}

export class Usuario2 {
        constructor(
           public id: number,
           public nombreUsuario: string,
           public correo: string,
           public token: string,
           public estatus: string,
           public contrasena: string,
           public rol: number
         ){}
}

export class UsuarioLDAP {
    constructor(
        public id: number,
        public nombreUsuario: string,
        public correo: string,
        public token: string,
        public estatus: string,
        public contrasena: string,
        public rol: number
      ){}
}
export class UsuarioCorreo {
  constructor(
    public id: number,
    public nombre: string,
    public token: string,
    public correo: string,
    public estatus: string
  ){}
}
export class Usuario3{
  constructor(
     public id: number,
     public nombre: string,
     public correo: string,
     public token: string,
     public estatus: string,
   ){}
}